src/gsIeti/gsIetiMapper.h

```diff
@@ -0,0 +1,176 @@
+    /// index and the vector representing the primal constraint.
+    void customPrimalConstraints( std::vector< std::pair<index_t,SparseVector> > data );
+
+    /// @brief This function constructs the global solution from a vector of patch-local ones
+    Matrix constructGlobalSolutionFromLocalSolutions( const std::vector<Matrix>& localContribs );
+
+    /// @brief This function returns a list of dofs that are (on the coarse level) coupled
+    ///
+    /// @param patch   Number of the patch
+    std::vector<index_t> getSkeletonDofs( index_t patch ) const;
+
+public:
+    /// @brief Returns the number of Lagrange multipliers.
+    index_t nLagrangeMultipliers()
+    {
+        GISMO_ASSERT(! m_jumpMatrices.empty(), "gsIetiMapper: Number of Lagrange multipliers not yet known.");
+        return m_jumpMatrices[0].rows();
+    }
+
+    /// @brief Returns the size of the primal problem (number of primal dofs)
+    index_t nPrimalDofs() const                                            { return m_nPrimalDofs;                }
+
+    /// @brief Returns the primalConstraints (as vectors) for the given patch
+    ///
+    /// These vectors form the matrix \f$ C_k \f$ in the local saddle point system, cf.
+    /// the documentation \a gsPrimalSystem
+    const std::vector<SparseVector> & primalConstraints(index_t k) const   { return m_primalConstraints[k];       }
+
+    /// @brief Returns the indices of the primal dofs that are associated to the primal constraints for the given patch
+    const std::vector<index_t> & primalDofIndices(index_t k) const         { return m_primalDofIndices[k];        }
+
+    /// @brief Returns the jump matrix \f$ B_k \f$ for the given patch
+    const JumpMatrix& jumpMatrix(index_t k) const                          { return m_jumpMatrices[k];            }
+
+    /// @brief The global dof mapper
+    const gsDofMapper& dofMapperGlobal() const                             { return m_dofMapperGlobal;            }
+
+    /// @brief The dof mapper for the given patch
+    const gsDofMapper& dofMapperLocal(index_t k) const                     { return m_dofMapperLocal[k];          }
+
+    /// @brief The function values for the eliminated dofs on the given patch
+    const Matrix& fixedPart(index_t k) const                               { return m_fixedPart[k];               }
+
+private:
+    const gsMultiBasis<T>*                        m_multiBasis;
+    gsDofMapper                                   m_dofMapperGlobal;
+    std::vector<gsDofMapper>                      m_dofMapperLocal;
+    std::vector<Matrix>                           m_fixedPart;
+    std::vector<JumpMatrix>                       m_jumpMatrices;
+    index_t                                       m_nPrimalDofs;
+    std::vector< std::vector<SparseVector> >      m_primalConstraints;
+    std::vector< std::vector<index_t> >           m_primalDofIndices;
+    unsigned                                      m_status;
```

Comment:
would be nice to describe each member with a one-liner doxy-comment here as well.

Reply:
Comments included